client/auth/index.js

```diff
@@ -9,10 +9,12 @@ import page from 'page';
 import config from 'config';
 import controller from './controller';
 
+import { makeLayout, render as clientRender } from 'controller';
+
 module.exports = function() {
 	if ( config.isEnabled( 'oauth' ) ) {
-		page( '/login', controller.login );
-		page( '/authorize', controller.authorize );
-		page( '/api/oauth/token', controller.getToken );
+		page('/login', controller.login, makeLayout, clientRender);
```

Comment:
That's sort of how we do isomorphic rendering of the theme showcase -- which is what I wrote the [`clientRouter`](https://github.com/Automattic/wp-calypso/blob/361aa5843e4e3e7a62d21e1e65516dfa0bda8e28/client/controller/index.web.js#L66) for, among other things :slightly_smiling_face: 

Here's an example:
https://github.com/Automattic/wp-calypso/blob/41f649dec0b196d76b02ecfdc1499ff6ed042efe/client/my-sites/theme/index.web.js#L9-L13
and the same on the server:
https://github.com/Automattic/wp-calypso/blob/41f649dec0b196d76b02ecfdc1499ff6ed042efe/client/my-sites/theme/index.node.js#L9-L13

In short, we added some trickery to `server/bundler` to pass a `router` argument to each section's entry point module, which for client-side rendering is [exactly what you found](https://github.com/Automattic/wp-calypso/blob/361aa5843e4e3e7a62d21e1e65516dfa0bda8e28/client/controller/index.web.js#L66) (`page`, more or less), and for server-side rendering is something a little more complicated, based on `express`.

(AFAICS the only difference to your suggestion is that we're still explicitly adding `makeLayout` to the route def. This is by design -- we want to make `context.layout` the actual single-tree entry point, while `primary` and `secondary` are thought of as just one possible way to implement that, with `makeLayout` being the adapter.)

But yeah, we could use this here. I was a bit on the fence since it would mean an even bigger change. Also, I was never quite sure if it was a good idea to implicitly render at the end of each route definition, since that conflicts with [a particular kind of route defs](https://github.com/visionmedia/page.js#separating-concerns). While I don't think we have many of these in Calypso, it'd be nice to not break them.

Finally, we might have to define all sections as `isomorphic` in `wordpress-com.js` for the relevant logic in `server/bundler` to work, and that might give us a number of other problems (Calypso might try to SSR everything then which clearly won't work, so we'd need to change some other pieces of logic.)

So I was rather thinking to drop `page` in favor of `router` in a later PR. But I'm happy to discuss and maybe change my mind, since none of my arguments seem all that compelling 😁